ToNote/Controls/ExtendedRichTextBox.cs

```diff
@@ -9,21 +11,77 @@ public class ExtendedRichTextBox : RichTextBox
     {
         public ExtendedRichTextBox()
         {
+            var range = new TextRange(Document.ContentStart, Document.ContentEnd);
+
             // Checks if backspace was pressed when the textbox was empty and raises an event. Used for convenient empty textbox removal
             this.PreviewKeyDown += (s, e) =>
             {
                 if (e.Key == System.Windows.Input.Key.Back)
                 {
-                    var range = new TextRange(Document.ContentStart, Document.ContentEnd);
                     if (string.IsNullOrWhiteSpace(range.Text))
                         BackspacePressedWhileEmpty?.Invoke(this, new RoutedEventArgs());
                 }
             };
 
+            this.TextChanged += (s, e) =>
+            {
+                //Prevents start of tracking while initializing.
+                if (!this.IsLoaded) return;
+
+                // -3, because text ending has \r\n chars
+                var offset = range.Text.Length - 3;
+                var secondToLastChar = offset - 1 >= 0 ? range.Text[offset - 1] : ' ';
+                var lastChar = offset >= 0 ? range.Text[offset] : ' ';
+
+                if (_tracking)
+                {
+                    //Stops tracking after 10 text changes without a '/' being the last character.
+                    if (_trackingCounter++ > 10)
+                    {
+                        _tracking = false;
+                    }
+
+                    //If text is deleted beyond '/', stops tracking
+                    if (offset - _slashIndex < 0)
+                    {
+                        _tracking = false;
+                        return;
+                    }
+
+                    var keyword = range.Text.Substring(_slashIndex + 1, offset - _slashIndex);
+
+                    foreach (var keywordAction in _trackedKeywords)
+                    {
+                        if (keyword != keywordAction.Keyword + ' ' && keyword != keywordAction.Keyword + '\r' + '\n') continue;
+
+                        _tracking = false;
+
+                        //Trims the '/' + keyword from the end of the text
+                        range.Text = range.Text.Remove(offset - keyword.Length, keyword.Length + 1);
+
+                        keywordAction.Action.Invoke();
+
+                        return;
```

Comment:
Remove this return, as you currently allow registering multiple keywords per action, but only execute the first found action.